src/shared/components/query/DataTypePrioritySelector.tsx

```diff
@@ -2,40 +2,60 @@ import * as _ from 'lodash';
 import * as React from 'react';
 import styles from './styles/styles.module.scss';
 import { QueryStore, QueryStoreComponent } from './QueryStore';
-import { toJS } from 'mobx';
 import { observer } from 'mobx-react';
-import { FlexRow, FlexCol } from '../flexbox/FlexBox';
+import { FlexRow } from '../flexbox/FlexBox';
 import SectionHeader from '../sectionHeader/SectionHeader';
-import { AlterationTypeConstants } from '../../../pages/resultsView/ResultsViewPageStore';
 import LoadingIndicator from 'shared/components/loadingIndicator/LoadingIndicator';
-import { MolecularProfile } from 'cbioportal-ts-api-client';
+import { MakeMobxView } from '../MobxView';
+import { remoteData } from 'cbioportal-frontend-commons';
+import { getMolecularProfileOptions } from './QueryStoreUtils';
 
 @observer
 export default class DataTypePrioritySelector extends QueryStoreComponent<
     {},
     {}
 > {
+    readonly molecularProfileCategorySet = remoteData({
+        await: () => [this.store.groupedMolecularProfilesByType],
+        invoke: () => {
+            return Promise.resolve(
+                getMolecularProfileOptions(
+                    this.store.groupedMolecularProfilesByType.result
+                )
+            );
+        },
+        default: [],
+    });
+
+    readonly flexRowContent = MakeMobxView({
+        await: () => [this.molecularProfileCategorySet],
+        render: () => {
+            return (
+                <div className="clearfix" style={{ display: 'flex' }}>

```

Comment:
i tihnk we can get rid of clearfix since we are now using flexbox.  clearfix was an old hack to make a container expand around floated contents